broadinstitute/pearl
# [JN-440] fine-grained environment copying

This adds per-config variable and per-survey/consent checkboxes for copying over configuration from one environment to another.  It also adds some sensible defaults -- if the target environment isn't initialized, everything will be selected by default.  Otherwise, everything will be unselected, since it's generally preferable to not copy things by accident than to accidentally copy over a draft that isn't IRB approved.  Still tons of room to improve the UX on this page (e.g. select all, clear all buttons, tooltips, confirm dialogs), but since this is just an internal tool for now (and we probably won't actually use it until August when the next round of OurHealth changes goes through, this should be good enough.

![image](https://github.com/broadinstitute/juniper/assets/2800795/97db8803-3289-4685-aa6e-94ddc4305de0)


TO TEST:
1. repopulate OurHealth (just to make sure you're starting fresh)
2. go to https://localhost:3000/ourhealth/env/irb/diff/sandbox
3. confirm you see a screen showing the differences between the uninitialized sandbox and initialized IRB environment, all boxes should be checked
4. click "Copy changes"
5. click into the sandbox environment, and update some surveys or change some configs
6. go back to https://localhost:3000/ourhealth/env/irb/diff/sandbox
7. confirm you see a screen showing the changes you made, with options to copy them (all deselected)
8. select some of the chagnes
9. hit "Copy changes"
10. go into the IRB environment and confirm the changes you selected are applied, and other changes are not
